Transactions.md

```diff
@@ -1,182 +1,291 @@
-  * Local Final Address: `BitcoinAddress`
-  * nLockTime: `UInt32`
-  * Local Payout: `CurrencyUnit`
-  * ToLocalOutput: `ScriptPubKey`
-  * Fee Rate: `FeeUnit`
-
-Where
-  - `ToLocalOutput` is of the form [specified above](#CETOutputs)
-### <a name="ClosingGlobal">Global</a>
-  * nLockTime is `0`
-### <a name="ClosingInputs">Inputs</a>
-  * Input Spending(P2WSH(ToLocalOutput))
-      * nSequence is `0xffffffff`
-### <a name="ClosingOutputs">Outputs</a>
-  * One output corresponding to `Local Final Address` with value `Local Payout`
-
-## <a name="ClosingPenalty">Closing Transaction (Penalty)</a>
-### <a name="ClosingKnownValues">Known Values</a>
-  * Local Address: `BitcoinAddress`
-  * Remote's ToLocalOutput: `ScriptPubKey`
-  * Fee Rate: `FeeUnit`
-
-Where
-  - `Local Address` is any unused local address
-  - `Remote's ToLocalOutput` is of the form [specified above](#CETOutputs)
-### <a name="ClosingInputs">Inputs</a>
-  * Input Spending(P2WSH(Remote's ToLocalOutput))
-### <a name="ClosingOutputs">Outputs</a>
-  * One output corresponding to `LocalAddress` with value `P2WSH(Remote's ToLocalOutput).value - fee`
+This details the exact format of all on-chain transactions, which both sides need to agree on to ensure signatures are valid.
 
+# Table of Contents
+
+* [Funding Transaction](#funding-transaction)
+  * [Funding Transaction Input and Output Ordering](#funding-transaction-input-and-output-ordering)
+  * [Funding Inputs](#funding-inputs)
+  * [FundingOutput](#funding-output)
+  * [Change Outputs](#change-outputs)
+* [Contract Execution Transaction](#contract-execution-transaction)
+  * [CET Outputs](#cet-outputs)
+    * [Offerer Output](#offerer-output)
+    * [Accepter Output](#accepter-output)
+* [Refund Transaction](#refund-transaction)
+* [Fees](#fees)
+  * [Expected Weight of the Funding Transaction](#expected-weight-of-the-funding-transaction)
+  * [Expected Weight of the Contract Execution or Refund Transaction](#expected-weight-of-the-contract-execution-or-refund-transaction)
+* [Test Vectors](#test-vectors)
+* [References](#references)
+* [Authors](#authors)
+
+# Funding Transaction
+
+* version: 2
+* locktime: 0
+
+The funding inputs and change output script public keys are negotiated in the offer and accept messages.
+
+## Funding Transaction Input and Output Ordering
+
+The Offerer/Initiator's inputs come first and in the order that they were listed in the Offer message, the Accepter's inputs come after this in the order they are listed in the Accept message. The funding output comes first, followed by the Offerer's change output if applicable, and then the Accepter's change output if applicable.
+
+Note that this will likely change in the future.
+
+## Funding Inputs
+
+All funding inputs must be Segwit or nested P2SH(Segwit) in order to protect against malleability attacks.
+
+## Funding Output
+
+* The funding output script is a P2WSH to:
+
+```
+2 <offer_funding_pubkey> <accept_funding_pubkey> 2 OP_CHECKMULTISIG
```

Comment:
Maybe we can sort these lexicographically, it should be an easy compute for each party + doesn't reveal which key is being used by each party. IIRC there was some bitcoin proposal to have all `OP_CHECKMULTISIG` keys sorted this way before.

Reply:
Looks like this is what LN does, would definitely be good to do this then so we can share anon set with them better.

https://github.com/lightningnetwork/lightning-rfc/blob/master/03-transactions.md#funding-transaction-output

Reply:
I think sorting need to be done as well.

Reply:
Turns out this is a BIP

https://github.com/bitcoin/bips/blob/master/bip-0067.mediawiki